internal/host/hostcommands/install_cmd.go

```diff
@@ -287,7 +293,19 @@ func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host, inve
 		}
 	}
 
-	if !hasStaticNetwork {
+	// Check if Manufacturer is IBM/S390 and ProductName is not "KVM/Linux" (the case for z/VM and LPAR).
+	// If this is the case than we need to extract the necessary z/VM kargs and append them.
+	isInventorySet = (inventory != nil && inventory.SystemVendor != nil)
```

Comment:
Basically at this point inventory could not be nil (at least what I now so far) because it would fail in some previous steps. SystemVendor might be nil because a system could not be detected properly. The detection rely on a third party tool (github.com/jaypipes/ghw) not supporting all architectures. I put in this condition because of failing test cases in install_cmd_test testing only a very small subset. So to prevent upcoming test cases to fail I put this check into my code part rather than change or adapt test cases.

Reply:
It's better the tests change to ensure inventory is non-nil rather the code adjusting its behavior to accommodate existing wrong testing behavior